include/fast_float/ascii_number.h

```diff
@@ -350,46 +363,50 @@ parse_number_string(UC const *p, UC const *pend,
     loop_parse_if_eight_digits(p, pend, i);
 
     while ((p != pend) && is_integer(*p)) {
-      uint8_t digit = uint8_t(*p - UC('0'));
-      ++p;
+      uint8_t const digit = uint8_t(*p - UC('0'));
       i = i * 10 + digit; // in rare cases, this will overflow, but that's ok
+      ++p;
     }
-    exponent = before - p;
-    answer.fraction = span<UC const>(before, size_t(p - before));
+    exponent = int32_t(before - p);
+    answer.fraction = span<UC const>(before, uint32_t(p - before));
     digit_count -= exponent;
   }
+#ifndef FASTFLOAT_ONLY_POSITIVE_C_NUMBER_WO_INF_NAN
   FASTFLOAT_IF_CONSTEXPR17(basic_json_fmt) {
     // at least 1 digit in fractional part
     if (has_decimal_point && exponent == 0) {
       return report_parse_error<UC>(p,
                                     parse_error::no_digits_in_fractional_part);
     }
   }
+#endif
   else if (digit_count == 0) { // we must have encountered at least one integer!
     return report_parse_error<UC>(p, parse_error::no_digits_in_mantissa);
   }
-  int64_t exp_number = 0; // explicit exponential part
-  if ((uint64_t(fmt & chars_format::scientific) && (p != pend) &&
-       ((UC('e') == *p) || (UC('E') == *p))) ||
-      (uint64_t(fmt & detail::basic_fortran_fmt) && (p != pend) &&
-       ((UC('+') == *p) || (UC('-') == *p) || (UC('d') == *p) ||
```

Comment:
@lemire @dalle Can you confirm that code is correct? I didn't thoroughly understand logic here.